tests/nest_tests/print_function_code_generator_test.py

```diff
@@ -69,7 +69,7 @@ def test_simple_print_statment(self):
 
         with open(str(os.path.realpath(os.path.join(os.path.dirname(__file__), os.path.join(
                 os.pardir, 'target', 'simple_print_test.cpp')))), 'r') as reader:
-            self.assertEqual(reader.read().count('std::cout'), 3)
+            self.assertEqual(reader.read().count('std::cout'), 5)
```

Comment:
Why are the `std::cout` counts increased here?

Reply:
To accommodate the extra ``std::cout``s that were added to the template.

Reply:
They get rendered into the ``.cpp`` file, still enclosed by the ``#ifdef DEBUG`` condition. They will only be ignored once they get to the C++ preprocessor.

Reply:
Yeah right, the generated code will still have it but will be ignored during runtime depending on the `#DEBUG` flag. Sorry for the confusion, my bad!